js-packages/@prestojs/viewmodel/src/ViewModelCache.ts

```diff
@@ -1228,14 +1228,23 @@ export default class ViewModelCache<
         const isCompound = Array.isArray(pkFieldName);
         // typescript failed to narrow pkFieldName to array using isCompound here
         // for some reason...
-        if (Array.isArray(pkFieldName) && typeof pk !== 'object') {
+        if (Array.isArray(pkFieldName) && typeof pk !== 'object' && typeof pk !== 'string') {
```

Comment:
right now this will work because the cache key for a PK is then used directly below _and_ the key includes the names of the primary keys... but this could change (the key doesn't actually need to include this - it's an implementation detail). I'd like to avoid supporting this format in the interface for `get` and leave it as is.

The alternative fix is pretty simple - in `getAll` just change the loop to iterate over the caches instead of keys and extract the PK value from there (which will be an object if it's a compound key):

```
for (const recordCache of this.cache.values()) {
            const pk = recordCache.recordPk;
```

the rest should just work as is

Reply:
OK fixed